crates/zed/src/zed/migrate.rs

```diff
@@ -100,6 +104,27 @@ impl MigrationBanner {
             }
         }
     }
+
+    fn update_banner_text(&mut self, cx: &mut Context<Self>) {
+        let file_type = match self.migration_type {
+            Some(MigrationType::Keymap) => "keymap",
+            Some(MigrationType::Settings) => "settings",
+            None => return,
+        };
+
+        let backup_file_name = self.backup_file_name();
+        let migration_text = format!(
+            "Your {} file uses deprecated settings which can be \
+            automatically updated. A backup will be saved to `{}`",
+            file_type,
+            format!("\u{00A0}{}\u{00A0}", backup_file_name)
```

Comment:
I suppose you added this to prevent whitespace wrapping, but it looks like it's still happening for end whitespace at `settings.json`. This seems like a GPUI bug, but for now I think it's okay to have no whitespace at all in the codeblock. That's what we do in other places.

```suggestion
            backup_file_name)
```